src/classes/BDI_Donations.cls

```diff
@@ -506,6 +508,15 @@ public class BDI_Donations {
             } else if (dataImport.PaymentImportStatus__c == System.Label.bdiMatchedNone) {
                 dataImport.PaymentImportStatus__c = bdi.statusMatchedNone();
             }
+
+            // campaigns -- list is shorter since we're limiting matching behavior
```

Comment:
Consider breaking the three sections of this method into three methods - `getCampaignImportStatus(String labelValue)`, etc.

Reply:
Also, is this really the best way to do this? Set status THEN see if we're running dry run then change it?

Reply:
That's a great question re: the latter part, and I think this written to avoid additional complexity during the writing of this code, or as a catch-all. I definitely would rather not rework the first two sections, but would be comfortable changing the campaigns section if you'd like to see that fixed here. Otherwise, I can break them out into separate methods if that's preferable.

Reply:
Let's take the opportunity to improve code where we can.

Reply:
I think this code is equivalent and avoids repeating the ifs:

```
Map<String, String> importStatusToDryRunStatus = new Map<String, String>{
	BDI_DataImport_API.bdiFailed => bdi.statusDIFailed(),
	System.Label.bdiMatched => bdi.statusMatchedExisting(),
	System.Label.bdiMatchedBest => bdi.statusMatchedBest(),
	System.Label.bdiMatchedNone => bdi.statusMatchedNone(),
	System.Label.bdiMatchedId => bdi.statusMatchedById()
}

private void setDryRunDonationStatuses(List<DataImport__c> dataImportDonations) {

    for (DataImport__c dataImport : dataImportDonations) {
        // opps
        dataImport.DonationImportStatus__c = importStatusToDryRunStatus.get(dataImport.DonationImportStatus__c);

        // payments
        dataImport.PaymentImportStatus__c = importStatusToDryRunStatus.get(dataImport.PaymentImportStatus__c);

        // campaigns -- list is shorter since we're limiting matching behavior
        dataImport.DonationCampaignImportStatus__c = importStatusToDryRunStatus.get(dataImport.DonationCampaignImportStatus__c);
    }
}
```

Reply:
Love it! I'll test this out to be sure, and will clean this up. Great suggestion!

Reply:
So I did quite a bit of testing, and for some reason putting importStatusToDryRunStatus outside of the setDryRunDonationStatuses as a private or public class level variable caused all kinds of test failures. I've wrapped it into this method in this next commit, and think it is still clearer to read than before. I also needed to factor in a null check, since we're not using else blocks in the code and need to be prepared for an import status not being in the map (there was exactly 1 test failure that helped catch this!).

Reply:
Yay for helpful tests!